src/worker.js

```diff
@@ -11,7 +11,6 @@
 var threadInfoStruct = 0; // Info area for this thread in Emscripten HEAP (shared). If zero, this worker is not currently hosting an executing pthread.
 var selfThreadId = 0; // The ID of this thread. 0 if not hosting a pthread.
 var parentThreadId = 0; // The ID of the parent pthread that launched this thread.
-var tempDoublePtr = 0; // A temporary memory area for global float and double marshalling operations.
```

Comment:
this is used on line 90, isn't it?

Reply:
Nah, that's `Module['tempDoublePtr']` and `e.data.tempDoublePtr`.
Unless I'm missing something, neither of those are the `tempDoublePtr` defined in the worker.

Reply:
Hmm, line 90 has `{{{ makeAsmGlobalAccessInPthread('tempDoublePtr') }}} = e.data.tempDoublePtr;`, and that will turn into either
```
tempDoublePtr = e.data.tempDoublePtr;
```
or if MODULARIZE,
```
Module['tempDoublePtr')] = e.data.tempDoublePtr;
```
The former case does reference that global.

But thinking more, it looks like maybe it shouldn't, though, if that's the only reference - nothing sets it. Thinking yet more, this is not going to matter with the upstream backend anyhow (which doesn't use that special global).

Anyhow, the change as is will write to a JS global (if not modularize), after removing the `var`, which I think we should avoid as it would be a regression. Probably that line should always write to `Module['tempDoublePtr']` (avoiding `makeAsmGlobalAccessInPthread`), unless maybe the line should only execute in modularize mode, but that sound wrong to me.

Reply:
I recently changed this in #9026. You might want to merge in new changes.

Reply:
Ah, you're right. Sounds like tempDoublePtr should only be available under !WASM_BACKEND && !MODULARIZE.
Thanks quantum5, I've merged your changes in.